browser-test/src/applicant/northstar_upsell.test.ts

```diff
@@ -0,0 +1,41 @@
+import {expect, test} from '../support/civiform_fixtures'
+import {
+  enableFeatureFlag,
+  loginAsAdmin,
+  loginAsTestUser,
+  logout,
+  validateAccessibility,
+} from '../support'
+
+test.describe('Upsell tests', {tag: ['@northstar']}, () => {
+  const programName = 'Sample program'
+
+  test.beforeEach(async ({page, adminPrograms, applicantQuestions}) => {
+    await loginAsAdmin(page)
+
+    await test.step('Setup: Publish program as admin', async () => {
+      await adminPrograms.addProgram(programName)
+      await adminPrograms.publishProgram(programName)
+      await adminPrograms.expectActiveProgram(programName)
+      await logout(page)
+    })
+
+    await loginAsTestUser(page)
+
+    await enableFeatureFlag(page, 'north_star_applicant_ui')
+
+    await test.step('Setup: submit application', async () => {
+      await applicantQuestions.clickApplyProgramButton(programName)
+      await applicantQuestions.submitFromReviewPage(true)
+    })
+  })
+
+  test('view application submitted page', async ({page}) => {
```

Comment:
you can also add a screenshot test (using validateScreenshot)

        await validateScreenshot(
          page,
          'upsell-north-star',
          /* fullPage= */ true,
          /* mobileScreenshot= */ true,
        )

Reply:
What value would a screenshot test bring here? I ask because we're already verifying that elements exist.

I think having one screenshot test for the page is fine, I'm just worried about over-testing. Screenshot tests will have a lot of spurious diffs as we update the page (which is going to happen a lot over the next few months).

[go/tott/710](https://goto.google.com/tott/710)

Reply:
the screenshot test would be there to make sure the page itself looks right. The biggest value is preventing inadvertent UI changes. Imagine a scenario when we make a change to the header that requires a new variable, but we forget to pass that variable through for the application confirmation page. It would mess up the rendering of the page, but without a screenshot we wouldn't catch it. It also has some value in validating that things look reasonable on mobile.

I think you're right that it's a little noisy and mildly annoying to have to update, but it's pretty lightweight to update and gives us pretty broad coverage for not a lot of effort